qemu/tests/nic_opt.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from virttest import utils_net
+from virttest import utils_test
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    NIC option check test.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def test_netperf():
+        """
+        Netperf stress test for nic option.
+        """
+        netperf_client = params.get("netperf_client", "main_vm")
+        netperf_server = params.get("netperf_server", "localhost")
+        sub_type = params.get("sub_type", "netperf_stress")
+        params["netperf_client"] = netperf_client
+        params["netperf_server"] = netperf_server
+        error_context.context("Run netperf stress test, client: %s, server: %s" %
+                              (netperf_client, netperf_server), logging.info)
+        utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+    def test_ping():
+        """
+        Ping test for nic option.
+        :return:
+        """
+        package_sizes = [int(i) for i in params.get("ping_package_sizes").split()]
+        if params["os_type"] == "windows":
+            ifname = utils_net.get_windows_nic_attribute(session=session,
+                                                         key="netenabled", value=True,
```

Comment:
ping -S <source ip>, the name is not valid, if it only with one network device, not need add "-S" for this case